services/spar/src/Spar/Scim/User.hs

```diff
@@ -610,6 +605,65 @@ assertHandleNotUsedElsewhere hndl uid = do
   unless ((userHandle =<< musr) == Just hndl) $
     assertHandleUnused' "userName does not match UserId" hndl uid
 
+-- | Helper function that given a brig user, creates a scim user on the fly or returns
+-- an already existing scim user
+synthesizeStoredUser :: TeamId -> User -> Scim.ScimHandler Spar (Scim.StoredUser SparTag)
+synthesizeStoredUser tid usr = do
+  let readState :: Spar (RichInfo, AccountStatus, UTCTime, UTCTime, URIBS.URI)
+      readState = do
+        richInfo <- Brig.getBrigUserRichInfo (BrigTypes.userId usr)
+        accStatus <- Brig.getStatus (BrigTypes.userId usr)
+        SAML.Time now <- SAML.getNow
+        (createdAt, lastUpdatedAt) <- fromMaybe (now, now) <$> wrapMonadClient (Data.readScimUserTimes (BrigTypes.userId usr))
+        baseuri <- asks $ derivedOptsScimBaseURI . derivedOpts . sparCtxOpts
+        pure (richInfo, accStatus, createdAt, lastUpdatedAt, baseuri)
+
+  let writeState :: UserId -> ManagedBy -> Scim.StoredUser SparTag -> Spar ()
+      writeState uid managedBy storedUser = do
+        wrapMonadClient $ Data.writeScimUserTimes storedUser
+        when (managedBy /= ManagedByScim) $ do
+          Brig.setBrigUserManagedBy uid ManagedByScim
+
+  (richInfo, accStatus, createdAt, lastUpdatedAt, baseuri) <- lift $ readState
+  storedUser <- synthesizeStoredUser' tid usr richInfo accStatus createdAt lastUpdatedAt baseuri
```

Comment:
we always write the time now. this is wrong! why do we even store the time if we always overwrite it with the latest?

Reply:
related: dc5f356  not sure if timestamps are correct now (if they are, the test suite is wrong).